tox.ini

```diff
@@ -54,19 +63,38 @@ commands =
 
 [testenv:unit]
 description = Run unit tests
+commands_pre =
+    poetry install --only main,charm-libs,unit --no-root
 commands =
-    poetry install --no-root --no-cache
-    poetry export --without-hashes -f requirements.txt -o requirements.txt
     poetry run coverage run --source={[vars]src_path} \
         -m pytest -v --tb native -s {posargs} {[vars]tests_path}/unit
     poetry run coverage report
 
-[testenv:integration-{lxd, k8s}-{kafka,mysql,mongodb,postgresql,opensearch}]
+[testenv:integration]
 description = Run integration tests
-commands =
-    poetry install --no-root --with dev --no-cache
-    # Should be in deps, but then the version is not kept strict but keep getting updated
-    pip install juju=={env:LIBJUJU_VERSION_SPECIFIER:2.9.42.4}
-    poetry export --without-hashes -f requirements.txt -o requirements.txt
-    poetry run pytest -v --tb native --log-cli-level=INFO -s --cloud={env:CLOUD} {posargs} {[vars]tests_path}/integration/{env:TEST_FILE}
+set_env =
+    {[testenv]set_env}
+    # Workaround for https://github.com/python-poetry/poetry/issues/6958
+    POETRY_INSTALLER_PARALLEL = false
+pass_env =
+    CI
+    GITHUB_OUTPUT
+    SECRETS_FROM_GITHUB
+allowlist_externals =
+    {[testenv:build-wrapper]allowlist_externals}
 
+    # Set the testing host before starting the lxd cloud
+    sudo
+    sysctl
+
+commands_pre =
+    poetry install --only integration --no-root
+
+    # Set the testing host before starting the lxd cloud
+    sudo sysctl -w vm.max_map_count=262144 vm.swappiness=0 net.ipv4.tcp_retries2=5
+
+    {[testenv:build-wrapper]commands_pre}
```

Comment:
Copied from the opensearch charm. I don't think I can inject it for a single test. Alternatively, I can try doing it with `subprocess` in the opensearch test's code.

Reply:
Opensearch can't deploy without it.

Reply:
Charms would still be built when running tests locally.